### PR TITLE
net-p2p/bitcoin-core: bump to 26.0

### PR DESCRIPTION
**Pull request by whitslack**
Differences versus 25.1:

* We're now fetching the sources from GitHub since that source would be harder to compromise than bitcoincore.org. See the [rationale][1].

* The Qt dependencies are bumped from 5.15.5 to 5.15.10 per upstream.

* The `dev-libs/libsecp256k1` dependency when `USE="system-libsecp256k1"` is bumped from 0.3.1 to 0.4.0 and now requires `USE="ellswift extrakeys"`.

* `configure.ac` is patched to check that the system-installed libsecp256k1 (if requested) indeed supports the required modules.

* Upstream dropped the syscall sandbox, so we no longer need to configure `--without-seccomp` to disable it.

* Upstream renamed the bash-completion scripts.

* When we're not cross-compiling, we now regenerate the example `bitcoin.conf` using `contrib/devtools/gen-bitcoin-conf.sh`, just in case upstream ever forgets to do it when tagging a release.

[1]: https://github.com/gentoo/gentoo/pull/33555#discussion_r1375507602

@thesamesam

### CODE REVIEW

**Review by Flowdalic — APPROVED** [review]
LGTM, but let's give @thesamesam some time for reviewing.

File: net-p2p/bitcoin-core/bitcoin-core-26.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+			 - Add your user(s) to the 'bitcoin' group.
+			 - Symlink ~/.bitcoin to /var/lib/bitcoind.
+		EOF
+	fi
+
+	if use berkdb ; then
+		# https://github.com/bitcoin/bitcoin/pull/28597
+		# https://bitcoincore.org/en/releases/26.0/#wallet
+		efmt ewarn <<-EOF
+			Creation of legacy (Berkeley DB) wallets is refused starting with this
```
Comment (Flowdalic) [inline net-p2p/bitcoin-core/bitcoin-core-26.0.ebuild:323]: I would replace "this release" with "version 26.0", as it reduces the changes required to the ebuild when bumping the package to a new verison.

Comment (whitslack) [reply net-p2p/bitcoin-core/bitcoin-core-26.0.ebuild]: Good catch. Fixed.

File: net-p2p/bitcoin-core/bitcoin-core-26.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+inherit autotools bash-completion-r1 check-reqs db-use desktop edo multiprocessing python-any-r1 systemd toolchain-funcs xdg-utils
+
+DESCRIPTION="Reference implementation of the Bitcoin cryptocurrency"
+HOMEPAGE="https://bitcoincore.org/"
+SRC_URI="https://github.com/bitcoin/bitcoin/archive/v${PV}.tar.gz -> ${P/-core}.tar.gz"
+S="${WORKDIR}/${P/-core}"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="amd64 ~arm ~arm64 ~ppc ~ppc64 x86 ~amd64-linux ~x86-linux"
```
Comment (Flowdalic) [inline net-p2p/bitcoin-core/bitcoin-core-26.0.ebuild:17]: This is a straight to stable commit. I will fix this on merge. But please tests your ebuilds with pkgcheck, which would have shown that.

Comment (whitslack) [reply net-p2p/bitcoin-core/bitcoin-core-26.0.ebuild:17]: Oops. Why wasn't that caught by the CI bot here?

> please tests your ebuilds with pkgcheck

I wish I could, but I honestly can't figure out how to run it.
```
$ pkgcheck scan net-p2p/bitcoin-core/bi
tcoin-core-26.0.ebuild
pkgcheck: error: failed parsing 'make.defaults' in '/etc/portage/profile': [Errno 2
1] Is a directory: '/etc/portage/profile/make.defaults'
```
Yes, no shit that's a directory. So? That's not supported? Crap tool. 👎🏻

**Comment by puchu** [comment]
the ebuild has a bug when compiling it with USE="-daemon"

/tmp/tmerge/portage/net-p2p/bitcoin-core-26.0/work/bitcoin-26.0/src/bitcoind not found or not executable.
 * ERROR: net-p2p/bitcoin-core-26.0::gentoo failed (compile phase):
 *   (no error message)
 * 
 * Call stack:
 *     ebuild.sh, line 136:  Called src_compile
 *   environment, line 3627:  Called die
 * The specific snippet of code:
 *       tc-is-cross-compiler || TOPDIR="${S}" bash contrib/devtools/gen-bitcoin-conf.sh || die;
 * 
 * If you need support, post the output of `emerge --info '=net-p2p/bitcoin-core-26.0::gentoo'`,
 * the complete build log and the output of `emerge -pqv '=net-p2p/bitcoin-core-26.0::gentoo'`.
 * The complete build log is located at '/tmp/portage/net-p2p:bitcoin-core-26.0:20231212-151603.log'.
 * For convenience, a symlink to the build log is located at '/tmp/tmerge/portage/net-p2p/bitcoin-core-26.0/temp/build.log'.
 * The ebuild environment file is located at '/tmp/tmerge/portage/net-p2p/bitcoin-core-26.0/temp/environment'.
 * Working directory: '/tmp/tmerge/portage/net-p2p/bitcoin-core-26.0/work/bitcoin-26.0'
 * S: '/tmp/tmerge/portage/net-p2p/bitcoin-core-26.0/work/bitcoin-26.0'

>>> Failed to emerge net-p2p/bitcoin-core-26.0, Log file:

**Comment by puchu** [comment]
the ebuild has a bug when compiling it with USE="-daemon"

/tmp/emerge/portage/net-p2p/bitcoin-core-26.0/work/bitcoin-26.0/src/bitcoind not found or not executable.
 * ERROR: net-p2p/bitcoin-core-26.0::gentoo failed (compile phase):
 *   (no error message)
 * 
 * Call stack:
 *     ebuild.sh, line 136:  Called src_compile
 *   environment, line 3627:  Called die
 * The specific snippet of code:
 *       tc-is-cross-compiler || TOPDIR="${S}" bash contrib/devtools/gen-bitcoin-conf.sh || die;
 * 
 * If you need support, post the output of `emerge --info '=net-p2p/bitcoin-core-26.0::gentoo'`,
 * the complete build log and the output of `emerge -pqv '=net-p2p/bitcoin-core-26.0::gentoo'`.
 * The complete build log is located at '/tmp/portage/net-p2p:bitcoin-core-26.0:20231212-151603.log'.
 * For convenience, a symlink to the build log is located at '/tmp/emerge/portage/net-p2p/bitcoin-core-26.0/temp/build.log'.
 * The ebuild environment file is located at '/tmp/emerge/portage/net-p2p/bitcoin-core-26.0/temp/environment'.
 * Working directory: '/tmp/emerge/portage/net-p2p/bitcoin-core-26.0/work/bitcoin-26.0'
 * S: '/tmp/emerge/portage/net-p2p/bitcoin-core-26.0/work/bitcoin-26.0'

>>> Failed to emerge net-p2p/bitcoin-core-26.0, Log file:

**Comment by Flowdalic** [comment]
Please fill a bug at bugs.gentoo.org.

**Pull request closed without merging**